pyravendb/store/document_store.py

```diff
@@ -100,7 +100,8 @@ def get_request_executor(self, db_name=None):
             raise ValueError("database name cannot be None")
 
         with self.lock:
-            self._request_executors.setdefault(db_name, RequestsExecutor.create(self.urls, db_name, self._certificate,
+            if self._request_executors.get(db_name) is None:
+                self._request_executors.setdefault(db_name, RequestsExecutor.create(self.urls, db_name, self._certificate,
                                                                                 self.conventions))
         return self._request_executors[db_name]
```

Comment:
You are still reading on one thread, modifying on the other, though.

Reply:
cc @gregolsky

Reply:
how about including the return statement inside the lock?